### ISSUE TITLE
[Modded] Issues with modded afflictions with the type "huskinfection" in status effects

### ISSUE THREAD

**Issue by Emanthealmighty** [Regalis11/Barotrauma#4935]
- [x] I have searched the issue tracker to check if the issue has already been reported.

**Description**
When using a function such as "conditional [affliction]="gte [number]" to check the strength of an affliction and do something accordingly, it will not work if the affliction in question is a type of modded husk infection that _doesn't_ override the normal husk infection, the conditional simply just fails to check the strength of the modded affliction.

Additionally, any affliction / reduceaffliction function that targets "huskinfection" will also target any affliction of the type "huskinfection" even if the function checks for the identifier and not the type, so, for example, calyxanide will work on every single modded husk infection even if it's not intended to.

**Version**
Version: 0.11.0.10 - Public Branch. OS: Windows 10.

**Comment by Regalis11** [comment]
I'm afraid I couldn't reproduce this. Tested by adding the following affliction and modifying calyxanide to target "moddedhuskinfection" instead of "huskinfection":

```
  <AfflictionHusk
    name="Modded Husk infection"
    identifier="moddedhuskinfection"
    description="Something dark and unpleasant moves in the mouth. They are rendered completely mute, save for occasional clicking sounds apparently emanating from deep within the throat."
    type="someinfection"
    targets="human,crawler"
    huskedspeciesname="[speciesname]husk"
    causeofdeathdescription="Taken over by a husk parasite"
    selfcauseofdeathdescription="You have been taken over by the husk parasite."
    limbspecific="false"
    indicatorlimb="Torso"
    activationthreshold="0"
    showiconthreshold="40"
    karmachangeonapplied="-1"
    maxstrength="100"
    achievementonremoved="healhusk"
    iconcolors="60,140,195,255;60,107,195,255;60,0,195,255">
    <Effect minstrength="0" maxstrength="100"
      maxvitalitydecrease="0"
      strengthchange="0.3"/>
    <icon texture="Content/UI/MainIconsAtlas.png" sourcerect="0,640,128,128" color="60,107,195,255" origin="0,0"/>
  </AfflictionHusk>
```

Are you using "huskinfection" as the custom affliction's type? Conditionals and affliction/reduceaffliction check if _either_ the type or the identifier of the affliction match, so if you're using the same type for both, things will get mixed up.

**Comment by Emanthealmighty** [comment]
When setting the type of an "afflictionhusk" affliction to something other than "huskinfection", the modded husk affliction will throw out errors when attempting to spawn a husk or spawn a husk appendage on a character. This also happens to me with the modded husk affliction you provided above.

**Comment by Regalis11** [comment]
Ah, you're right, should've tested this a bit more carefully! Fixed in https://github.com/Regalis11/Barotrauma-development/commit/328f37f5470903080cb2fa262b142d0bf81998d7

**Comment by Emanthealmighty** [comment]
Fixed as of version V0.1300.0.3 (Unstable)

**Comment by Rokvach** [comment]
Closing
